### PR TITLE
feat: implement tutorials sitemap

### PR DESCRIPTION
**Pull request by zchsh**
## ✅ "Ready for Review" checklist

- [x] The Vercel preview link has been added to this PR's description
- [x] The Asana task has been added to this PR's description
- [x] This PR's link has been added to the "📐 Pull Request" field on the Asana task
- [x] This Vercel preview link has been added to the "🔍 Preview" field on the Asana task
- [x] The description template has been filled in below

---

## 🔗 Relevant links

- [Preview link][waypoint-tutorials] 🔎
- [Asana task][task] 🎟️ 

## 🗒️ What

Implements the `Sitemap` display component for `views/product-tutorial-view`.

## 🤷 Why

To finish building out the `product-tutorial-view`.

## 🛠️ How

- Moves the sitemap component's heading into the component, to clean up JSX at the page level
- Add styles to the existing stubbed `sitemap` component

## 📸 Design Screenshots

<details>
<summary>⤵️ Waypoint</summary>

### Design

waypoint-design

### Implementation

waypoint-implementation

</details>

<details>
<summary>⤵️ Vault</summary>

### Design

vault-design

### Implementation

> Note: I think my screenshot utility had trouble with this, since it's so long. Text appears sqooshed in this screenshot; in the browser this is not the case.

vault-implementation

</details>

<details>

<summary>⤵️ Full design spec</summary>

sitemap-full-spec

</details>

## 🧪 Testing

- [ ] Visit [the Waypoint tutorials landing][waypoint-tutorials]
    - [ ] Ensure the `Sitemap` component looks as expected
- [ ] Visit [the Vault tutorials landing][vault-tutorials]
    - [ ] Ensure the `Sitemap` component looks as expected

## 💭 Anything else?

Not at the moment!

[waypoint-tutorials]: https://dev-portal-git-zstutorials-sitemap-hashicorp.vercel.app/waypoint/tutorials
[vault-tutorials]: https://dev-portal-git-zstutorials-sitemap-hashicorp.vercel.app/vault/tutorials
[task]: https://app.asana.com/0/1202097197789424/1202182325935202/f

### CODE REVIEW

File: src/views/product-tutorials-view/components/sitemap/sitemap.module.css [hunk 1]
```diff
@@ -1,4 +1,98 @@
...
+  margin-right: 8px;
+}
+
+.tutorialListRoot {
+  /* note: top & bottom margin adjusted for use of 4px padding in list items */
+  margin: 12px 0 -4px 0;
+  list-style: none;
+  padding: 0;
+
+  &.hasMultipleItems {
```
Comment (zchsh) [inline src/views/product-tutorials-view/components/sitemap/sitemap.module.css:54]: Some collections only have one tutorials, and this one tutorial may have a very long name. In these cases, it feels odd to force the tutorial into a half-width column, potentially causing text to wrap.

**Comment by zchsh** [comment]
@kendallstrautman @braydenlove This PR implements the sitemap component - let me know what ya'll think 🙇‍♂️

File: src/views/product-tutorials-view/components/sitemap/index.tsx [hunk 2]
```diff
@@ -1,34 +1,77 @@
...
-          </ul>
-        </li>
-      ))}
-    </ul>
+    <div className={s.root}>
+      <div className={s.headingBar}>
+        <HeadingElem id={slug} className={s.heading}>
+          {title}
+        </HeadingElem>
+        <StandaloneLink
```
Comment (kendallstrautman) [inline src/views/product-tutorials-view/components/sitemap/index.tsx:36]: 💭 Hmmm... I sort of feel like we should remove this for beta. Feels odd to be booting out back to Learn....But maybe we can just talk with Product / Design as we're doing polish. 

What are your thoughts @zchsh?

**Review by kendallstrautman — APPROVED** [review]
Looks great! Just a UX related comment on the link to external search. But we can sort that out beyond this PR.

File: src/views/product-tutorials-view/components/sitemap/index.tsx (hunk 2 above)
Comment (zchsh) [reply src/views/product-tutorials-view/components/sitemap/index.tsx:36]: My memory might be off, but I recall in some Figma sync at some point that we landed on keeping this for June 20 - but maybe with some analytics? The intent being to collect data on how much people would be looking for the "search all tutorials page"?

Fully agreed that it feels a little odd for beta though, with the assumption that I think we want to prioritize collecting feedback on the new experience, so try to keep folks who've opted in within that experience.

Definitely happy to drop this out, or add some `heap` tracking or something - either in this PR or a follow-up 👍

Comment (braydenlove) [reply src/views/product-tutorials-view/components/sitemap/index.tsx:36]: Sorry for the confusion, we got feedback from Adam and Geoffrey and decided to remove it all together.

Comment (zchsh) [reply src/views/product-tutorials-view/components/sitemap/index.tsx:36]: Sweet, will do now! 👍

**Comment by braydenlove** [comment]
This looks great, the only question I have is around the icon being part of the collection link. Should those be seperate? I could go either way but feel like it is a bit odd not to have a hover state if it is a link. Also curious about the lack of hover states on the links themselves. Otherwise great work per usual :)

File: src/views/product-tutorials-view/components/sitemap/index.tsx (hunk 2 above)
Comment (braydenlove) [reply src/views/product-tutorials-view/components/sitemap/index.tsx]: Sorry about that I discussed it with Brandon but should have ran it past you two as well. Will be sure to do that in the future!

**Comment by zchsh** [comment]
@braydenlove Great questions! I can definitely implement hover states for the links - I might've missed those in designs, what would we be looking at? A color shift maybe?

**Comment by braydenlove** [comment]
No worries they are a bit hidden! 
image
[Link](https://www.figma.com/file/VD7ahvXuXWJApeGnhbW4hv/Dev-Portal?node-id=8888%3A114658)

**Comment by braydenlove** [comment]
Let me know if you have questions @zchsh, happy to go through it on Monday as well.

**Comment by zchsh** [comment]
@braydenlove I'm 🤦 over here, thanks for calling out the `:hover` styles 💖 Have now got those implemented, so merging this in 👍

**Pull request merged**